Adding kubernetes-spring-starter causes RuntimeException During Build

When I add
```
<dependency>
                <groupId>io.ap4k</groupId>
                <artifactId>kubernetes-spring-starter</artifactId>
                <version>0.3.4</version>
            </dependency>
```

To my Spring Boot app and run `mvn clean package` it results in 

```
Caused by: java.lang.RuntimeException: Error writing resources
    at io.ap4k.processor.AptWriter.write (AptWriter.java:89)
    at io.ap4k.processor.AptWriter.lambda$write$1 (AptWriter.java:65)
    at java.lang.Iterable.forEach (Iterable.java:75)
    at io.ap4k.processor.AptWriter.write (AptWriter.java:65)
    at io.ap4k.Session.close (Session.java:116)
    at io.ap4k.kubernetes.apt.KubernetesAnnotationProcessor.process (KubernetesAnnotationProcessor.java:40)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor (JavacProcessingEnvironment.java:794)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$200 (JavacProcessingEnvironment.java:91)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs (JavacProcessingEnvironment.java:627)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run (JavacProcessingEnvironment.java:1033)
    at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing (JavacProcessingEnvironment.java:1198)
    at com.sun.tools.javac.main.JavaCompiler.processAnnotations (JavaCompiler.java:1170)
    at com.sun.tools.javac.main.JavaCompiler.compile (JavaCompiler.java:856)
    at com.sun.tools.javac.main.Main.compile (Main.java:523)
    at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
    at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
    at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:126)
    at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:169)
    at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:785)
    at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:129)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

@ryanjbaxter Thanks! I did manage to reproduce using your branch.

I am pasting the full stack trace of the issue:

    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project gateway: Fatal error compiling: java.lang.RuntimeException: Error writing resources: Attempt to reopen a file for path /home/iocanel/workspace/src/github.com/ryanjbaxter/chaos-monkey-spring-boot-demo/gateway/target/classes/META-INF/ap4k/.config/springapplication.yml -> [Help 1]
    org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project gateway: Fatal error compiling
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    Caused by: org.apache.maven.plugin.MojoExecutionException: Fatal error compiling
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:796)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:129)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    Caused by: org.codehaus.plexus.compiler.CompilerException: java.lang.RuntimeException: Error writing resources
        at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:191)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:169)
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:785)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:129)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    Caused by: java.lang.RuntimeException: java.lang.RuntimeException: Error writing resources
        at com.sun.tools.javac.main.Main.compile (Main.java:553)
        at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
        at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
        at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:126)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:169)
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:785)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:129)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    Caused by: java.lang.RuntimeException: Error writing resources
        at io.ap4k.processor.AptWriter.write (AptWriter.java:89)
        at io.ap4k.processor.AptWriter.lambda$write$1 (AptWriter.java:65)
        at java.lang.Iterable.forEach (Iterable.java:75)
        at io.ap4k.processor.AptWriter.write (AptWriter.java:65)
        at io.ap4k.Session.close (Session.java:116)
        at io.ap4k.kubernetes.apt.KubernetesAnnotationProcessor.process (KubernetesAnnotationProcessor.java:40)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor (JavacProcessingEnvironment.java:794)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$200 (JavacProcessingEnvironment.java:91)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs (JavacProcessingEnvironment.java:627)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run (JavacProcessingEnvironment.java:1033)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing (JavacProcessingEnvironment.java:1198)
        at com.sun.tools.javac.main.JavaCompiler.processAnnotations (JavaCompiler.java:1170)
        at com.sun.tools.javac.main.JavaCompiler.compile (JavaCompiler.java:856)
        at com.sun.tools.javac.main.Main.compile (Main.java:523)
        at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
        at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
        at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:126)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:169)
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:785)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:129)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    Caused by: javax.annotation.processing.FilerException: Attempt to reopen a file for path /home/iocanel/workspace/src/github.com/ryanjbaxter/chaos-monkey-spring-boot-demo/gateway/target/classes/META-INF/ap4k/.config/springapplication.yml
        at com.sun.tools.javac.processing.JavacFiler.checkFileReopening (JavacFiler.java:535)
        at com.sun.tools.javac.processing.JavacFiler.createResource (JavacFiler.java:431)
        at io.ap4k.processor.AptWriter.write (AptWriter.java:82)
        at io.ap4k.processor.AptWriter.lambda$write$1 (AptWriter.java:65)
        at java.lang.Iterable.forEach (Iterable.java:75)
        at io.ap4k.processor.AptWriter.write (AptWriter.java:65)
        at io.ap4k.Session.close (Session.java:116)
        at io.ap4k.kubernetes.apt.KubernetesAnnotationProcessor.process (KubernetesAnnotationProcessor.java:40)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor (JavacProcessingEnvironment.java:794)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$200 (JavacProcessingEnvironment.java:91)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.runContributingProcs (JavacProcessingEnvironment.java:627)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run (JavacProcessingEnvironment.java:1033)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing (JavacProcessingEnvironment.java:1198)
        at com.sun.tools.javac.main.JavaCompiler.processAnnotations (JavaCompiler.java:1170)
        at com.sun.tools.javac.main.JavaCompiler.compile (JavaCompiler.java:856)
        at com.sun.tools.javac.main.Main.compile (Main.java:523)
        at com.sun.tools.javac.api.JavacTaskImpl.doCall (JavacTaskImpl.java:129)
        at com.sun.tools.javac.api.JavacTaskImpl.call (JavacTaskImpl.java:138)
        at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess (JavaxToolsCompiler.java:126)
        at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile (JavacCompiler.java:169)
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute (AbstractCompilerMojo.java:785)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute (CompilerMojo.java:129)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
        at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
        at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
        at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
        at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
        at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke (Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

--- a/core/src/main/java/io/ap4k/Configurators.java
+++ b/core/src/main/java/io/ap4k/Configurators.java
@@ -20,20 +20,24 @@ import io.ap4k.kubernetes.config.Configuration;
 import io.ap4k.config.ConfigurationSupplier;
 import io.ap4k.kubernetes.config.Configurator;
 
+import java.lang.reflect.Type;
 import java.util.HashSet;
 import java.util.LinkedHashSet;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 public class Configurators {
 
-  private final Set<ConfigurationSupplier<? extends Configuration>> suppliers = new LinkedHashSet<>();
+  private final Map<Type, ConfigurationSupplier<? extends Configuration>> suppliers = new ConcurrentHashMap<>();
   private final Set<Configurator> configurators = new HashSet<>();
 
   public void add(ConfigurationSupplier supplier) {
-    this.suppliers.add(supplier);
+    this.suppliers.put(supplier.getType(), supplier);
   }
   /**
    * Add a {@link Configurator}.
@@ -45,6 +49,7 @@ public class Configurators {
 
   public Stream<? extends Configuration> stream() {
     return suppliers
+      .values()
       .stream()
       .map(s -> s.configure(configurators).get());
   }
@@ -57,3 +62,4 @@ public class Configurators {
     return stream().filter(i -> type.isAssignableFrom(i.getClass())).map(i -> (C)i).findFirst();
   }
 }
+

--- a/core/src/main/java/io/ap4k/config/ConfigurationSupplier.java
+++ b/core/src/main/java/io/ap4k/config/ConfigurationSupplier.java
@@ -22,6 +22,8 @@ package io.ap4k.config;
 import io.ap4k.deps.kubernetes.api.builder.VisitableBuilder;
 import io.ap4k.kubernetes.config.Configurator;
 
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
 import java.util.function.Supplier;
 
 /**
@@ -49,4 +51,12 @@ public class ConfigurationSupplier<C> implements Supplier<C> {
     builder.accept(configurator);
     return this;
   }
+
+  
+  public Type getType() {
+    Class builderClass = builder.getClass();
+    ParameterizedType parameterizedType = (ParameterizedType) builderClass.getGenericSuperclass();
+    return parameterizedType.getActualTypeArguments()[0];
+  }
+
 }

--- a/core/src/main/java/io/ap4k/processor/AptWriter.java
+++ b/core/src/main/java/io/ap4k/processor/AptWriter.java
@@ -86,7 +86,7 @@ public class AptWriter implements SessionWriter, WithProject {
         return new AbstractMap.SimpleEntry<>(yml.toString(), value);
       }
     } catch (IOException e) {
-      throw new RuntimeException("Error writing resources");
+      throw new RuntimeException("Error writing resources", e);
     }
   }
 
@@ -104,7 +104,7 @@ public class AptWriter implements SessionWriter, WithProject {
         return new AbstractMap.SimpleEntry<>(yml.toString(), value);
       }
     } catch (IOException e) {
-      throw new RuntimeException("Error writing resources");
+      throw new RuntimeException("Error writing resources", e);
     }
   }
 
@@ -130,7 +130,7 @@ public class AptWriter implements SessionWriter, WithProject {
         result.put(yml.getName(), yamlValue);
       }
     } catch (IOException e) {
-      throw new RuntimeException("Error writing resources");
+      throw new RuntimeException("Error writing resources", e);
     }
     return null;
   }

--- a/core/src/main/java/io/ap4k/processor/SimpleFileWriter.java
+++ b/core/src/main/java/io/ap4k/processor/SimpleFileWriter.java
@@ -68,7 +68,7 @@ public class SimpleFileWriter implements SessionWriter, WithProject {
         return new AbstractMap.SimpleEntry<>(yml.toString(), value);
       }
     } catch (IOException e) {
-      throw new RuntimeException("Error writing resources");
+      throw new RuntimeException("Error writing resources", e);
     }
   }
 
@@ -92,7 +92,7 @@ public class SimpleFileWriter implements SessionWriter, WithProject {
       }
 
     } catch (IOException e) {
-      throw new RuntimeException("Error writing resources");
+      throw new RuntimeException("Error writing resources", e);
     }
   }
 
@@ -134,7 +134,7 @@ public class SimpleFileWriter implements SessionWriter, WithProject {
 
       return result;
     } catch (IOException e) {
-      throw new RuntimeException("Error writing resources");
+      throw new RuntimeException("Error writing resources", e);
     }
   }
 }

--- a/core/src/main/java/io/ap4k/project/AptProjectFactory.java
+++ b/core/src/main/java/io/ap4k/project/AptProjectFactory.java
@@ -51,7 +51,7 @@ public class AptProjectFactory {
       f = environment.getFiler().createResource(StandardLocation.CLASS_OUTPUT, "", ".marker-" + UUID.randomUUID().toString());
       return FileProjectFactory.create(Paths.get(f.toUri()).toFile());
     } catch (IOException e) {
-      throw new RuntimeException("Failed to determine the project root!");
+      throw new RuntimeException("Failed to determine the project root!", e);
     } finally {
       if (f != null) {
         f.delete();

--- a/core/src/main/java/io/ap4k/project/MavenInfoReader.java
+++ b/core/src/main/java/io/ap4k/project/MavenInfoReader.java
@@ -73,11 +73,11 @@ public class MavenInfoReader implements BuildInfoReader {
       DocumentBuilder builder = factory.newDocumentBuilder();
       return builder.parse(pom.toFile());
     } catch (IOException e) {
-      throw new RuntimeException(("Failed to read: " + pom.toAbsolutePath()));
+      throw new RuntimeException(("Failed to read: " + pom.toAbsolutePath()), e);
     } catch (ParserConfigurationException e) {
-      throw new RuntimeException(("Failed to parse: " + pom.toAbsolutePath()));
+      throw new RuntimeException(("Failed to parse: " + pom.toAbsolutePath()), e);
     } catch (SAXException e) {
-      throw new RuntimeException(("Failed to parse: " + pom.toAbsolutePath()));
+      throw new RuntimeException(("Failed to parse: " + pom.toAbsolutePath()), e);
     }
   }
 

--- a/core/src/main/java/io/ap4k/utils/Serialization.java
+++ b/core/src/main/java/io/ap4k/utils/Serialization.java
@@ -258,7 +258,7 @@ public class Serialization {
       }
       return outputStream.toString();
     } catch (IOException e) {
-      throw new RuntimeException("Unable to read InputStream." + e);
+      throw new RuntimeException("Unable to read InputStream." , e);
     }
   }
 

